homeassistant/helpers/event.py

```diff
@@ -1480,7 +1480,10 @@ def async_call_at(
         if isinstance(action, HassJob)
         else HassJob(action, f"call_at {loop_time}")
     )
-    return hass.loop.call_at(loop_time, _run_async_call_action, hass, job).cancel
+    handle = hass.loop.call_at(loop_time, _run_async_call_action, hass, job)
+    if job.cancel_on_shutdown:
+        hass.async_track_timer_handle(handle)
+    return handle.cancel
```

Comment:
We could make this its own function and share it between here and `async_call_later`